Files/ViewModels/ItemViewModel.cs

```diff
@@ -1878,17 +1890,13 @@ public void AddItemsToCollectionAsync(string path, string previousDir)
             RapidAddItemsToCollectionAsync(path, previousDir);
         }
 
-        public async Task AddFolderAsync(StorageFolder folder, string dateReturnFormat)
+        public async Task<ListedItem> AddFolderAsync(StorageFolder folder, string dateReturnFormat)
         {
             var basicProperties = await folder.GetBasicPropertiesAsync();
 
-            if ((AssociatedInstance.ContentFrame.SourcePageType == typeof(GenericFileBrowser)) || (AssociatedInstance.ContentFrame.SourcePageType == typeof(GridViewBrowser)))
+            if (!_addFilesCTS.IsCancellationRequested && ((AssociatedInstance.ContentFrame.SourcePageType == typeof(GenericFileBrowser)) || (AssociatedInstance.ContentFrame.SourcePageType == typeof(GridViewBrowser))))
```

Comment:
Could you write this check as (personal preference && coherence with other similar checks):
```cs
if (_addFilesCTS.IsCancellationRequested)
{	
     return null;	
}
if ((AssociatedInstance.ContentFrame.SourcePageType == typeof(GenericFileBrowser)) || (AssociatedInstance.ContentFrame.SourcePageType == typeof(GridViewBrowser)))
{
    return new ....
}
```